### PR TITLE
feat(rollout): add continuous batching profiling

### PR DESCRIPTION
**Pull request by nathon-lee**
## Summary
Thanks @delock for the continuous batching prototype and profiling follow-up.
Continuous batching currently bypasses `HybridEngineRollout` profiling, so no useful profile is recorded. This PR adds opt-in profiling for continuous batching, covering:
- prefill and decode forward time/counts
- scheduler and cache-management overhead
- generated tokens and throughput
- active batch size and configured capacity
Profiling remains disabled by default and does not change generation outputs or scheduling behavior.
## Validation
- Focused rollout tests: `38 passed`
- `py_compile`: passed
- `git diff --check`: passed
- YAPF and flake8 checks: passed

Related to #8459

### CODE REVIEW

File: tests/unit/runtime/rollout/test_hybrid_engine_rollout.py [hunk 1]
```diff
@@ -265,6 +265,61 @@ def forward(self, input_ids, attention_mask, past_key_values=None, use_cache=Tru
     ]
 
 
+def test_continuous_generation_profile_on_cuda_device():
```
Comment (delock) [inline tests/unit/runtime/rollout/test_hybrid_engine_rollout.py]: if this test need to be tested on true GPU device, should move under v1 directory.  cuda usage should be replaced by `get_accelerator()`

**Comment by delock** [comment]
@nathon-lee thanks!  I have left my comments.
Do you have sample profiling output for a small model (i.e. 8B)?

File: tests/unit/runtime/rollout/test_hybrid_engine_rollout.py (hunk 1 above)
Comment (nathon-lee) [reply tests/unit/runtime/rollout/test_hybrid_engine_rollout.py]: Thanks @delock for the feedback. I moved the accelerator-dependent test under tests/unit/v1 and replaced the hardcoded CUDA usage with get_accelerator(). The focused tests now pass: 34 passed.

**Comment by nathon-lee** [comment]
> @nathon-lee thanks! I have left my comments. Do you have sample profiling output for a small model (i.e. 8B)?

Thanks @delock. I ran a real profiling example with Qwen3-8B on CUDA using two prompts and a continuous batch size of 2.

```json
{
  "active_batch_size": 2,
  "batch_size": 2,
  "cache_management_overhead_ms": 5.7515,
  "continuous_batch_size": 2,
  "decode_forward_ms": 2065.3628,
  "generation_ms": 2121.7179,
  "generation_overhead_ms": 27.1613,
  "num_decode_forwards": 63,
  "num_generated_tokens": 128,
  "num_prefill_forwards": 1,
  "num_samples_per_prompt": 1,
  "post_processing_ms": 0.1966,
  "prefill_forward_ms": 29.1938,
  "prompt_expansion_ms": 0.0,
  "prompt_length": 12,
  "response_length": 64,
  "scheduler_overhead_ms": 1.7312,
  "tokens_per_second": 60.3229,
  "total_ms": 2121.9139
}
```

The generated output was:

```text
[
  "Explain why GPU memory bandwidth matters for LLM inference. GPU memory bandwidth is the rate at which data can be read from or written to a GPU's memory. For large language model (LLM) inference, this",
  "Give three practical ways to reduce inference latency. Reducing inference latency is crucial for applications that require real-time or near-real-time performance, such as autonomous vehicles, chatbots, and real-time analytics."
]
```

**Review by delock — APPROVED** [review]
